packages/xml-views-validation/src/validators/document/non-unique-id.ts

```diff
@@ -0,0 +1,105 @@
+    duplicatedIdsRecords,
+    buildIssuesForSingleID
+  );
+
+  return allIDsIssues;
+}
+
+function buildIssuesForSingleID(
+  duplicatedAttributes: DuplicatedIDXMLAttribute[],
+  id: string
+): NonUniqueIDIssue[] {
+  const issuesForID = map(
+    duplicatedAttributes,
+    (currDupAttrib, currAttribIdx) => {
+      const currDupIdValToken = currDupAttrib.syntax.value;
+      // Related issues must not include the "main" issue attribute
+      const relatedOtherDupIDAttribs = reject(
+        duplicatedAttributes,
+        (_, arrIdx) => arrIdx === currAttribIdx
+      );
+
+      return {
+        kind: "NonUniqueIDIssue" as "NonUniqueIDIssue",
+        message: `Non-unique ID value: "${id}" found.`,
+        severity: "error" as "error",
+        offsetRange: {
+          start: currDupIdValToken.startOffset,
+          end: currDupIdValToken.endOffset,
+        },
+        identicalIDsRanges: map(relatedOtherDupIDAttribs, (_) => ({
+          start: _.syntax.value.startOffset,
+          end: _.syntax.value.endOffset,
+        })),
+      };
+    }
+  );
+
+  return issuesForID;
+}
+
+type DuplicatedIDXMLAttribute = XMLAttribute & { syntax: { value: XMLToken } };
+
+class IdsCollectorVisitor implements XMLAstVisitor {
+  public idsToXMLElements: Record<
+    string,
+    DuplicatedIDXMLAttribute[]
+  > = Object.create(null);
+
+  visitXMLAttribute(attrib: XMLAttribute): void {
+    if (
+      attrib.key === "id" &&
+      attrib.value !== null &&
+      attrib.value !== "" &&
+      attrib.syntax.value !== undefined &&
+      attrib.parent.name !== null &&
+      // Heuristic to limit false positives by only checking tags starting with upper
+      // case names, This would **mostly** limit the checks for things that can actually be
+      // UI5 Elements / Controls.
+      /^[A-Z]/.test(attrib.parent.name) &&
+      !isNoneUI5id(attrib)
+    ) {
+      if (this.idsToXMLElements[attrib.value] === undefined) {
+        // @ts-ignore - TSC does not understand: `attrib.syntax.value !== undefined` is a type guard
+        this.idsToXMLElements[attrib.value] = [attrib];
+      } else {
+        // @ts-ignore - TSC does not understand: `attrib.syntax.value !== undefined` is a type guard
+        this.idsToXMLElements[attrib.value].push(attrib);
+      }
+    }
+  }
+}
+
+// We only care about UI5 elements/controls IDs when check non-unique IDs
+// `id` attributes in these: **known** namespaces which are sometimes used
+// in UI5 xml-views are definitively not relevant for this validation
+const whiteListedNamespaces: Record<string, boolean> = {
+  "http://www.w3.org/1999/xhtml": true,
+  "http://www.w3.org/2000/svg": true,
+  "http://schemas.sap.com/sapui5/extension/sap.ui.core.template/1": true,
+};
+
+function isNoneUI5id(attrib: XMLAttribute): boolean {
+  const parentElement = attrib.parent;
+  const parentPrefix = parentElement.ns ?? DEFAULT_NS;
+  const parentResolvedNamespace = parentElement.namespaces[parentPrefix];
```

Comment:
You can use `resolveXMLNS` from `logic-utils` for this now

Reply:
ok